File: deprecated/api/PullCourseGrades/pull_grades.py
# This script is only a draft
import requests
import json
import pprint

# API reference https://canvas.instructure.com/doc/api/submissions.html

BASE_URL = "https://school.instructure.com/api/v1%s"
access_token = 'token_here'
course_id = 'change_me'  # not the sis_id but the canvas internal id

REQUEST_HEADERS = {'Authorization':'Bearer %s' % access_token}

# First, get the list of students in the course

students_endpoint = BASE_URL % '/courses/%s/students' % (course_id)

# Create a request, adding the REQUEST_HEADERS to it for authentication
not_done = True
students = []
url = students_endpoint
while not_done:
  student_request = requests.get(url,headers=REQUEST_HEADERS)
  students+=student_request.json()
  if 'next' in student_request.links.keys():
    url = student.request.links['next']['href']
  else:
    not_done = False

print 'done gettign students',len(students),'students'
# Load the response as JSON
response_data = student_request.json()

# Exit if there were no students in the returned data
if not response_data:
  print 'Sorry, there were no students registered in the course.'
  exit(0)

# Loop through the students, populating the student_ids list with their canvas ids
student_ids = [s['id'] for s in response_data]


# Build the endpoint for requesting submissions
submissions_endpoint = BASE_URL % '/courses/%s/students/submissions' % (course_id)

# Build the GET request parameters that are needed to fetch the submissions along with the
# total scores (grades)
submission_params = {'include[]':'total_scores','grouped':1}
submission_params['student_ids[]'] = student_ids

#submission_params = urllib.urlencode(submission_params)


# Build a request, adding the REQUEST_HEADERS to it for authentication
req = requests.get(submissions_endpoint, params=submission_params, headers=REQUEST_HEADERS)

# Load the response as JSON
grades = req.json()

pprint.pprint(grades)

File: deprecated/api/PullCourseGrades/pull_grades.php
<?php
/**
 * This script will fetch the current final graes for all students in a specific course. 
 * In order for this to work, you need to be either a teacher in the course or an account 
 * admin.  Please take a moment to customize the variables in the top part of this script 
 * to fit your situation.  Specifically, adjust BASE_URL, ACCESS_TOKEN, and $course_id.
 *
 **/

// Variables you should adjust
define('BASE_URL',"https://<subdomain>.instructure.com/api/v1%s"); # Set this to your account's sub-domain or url.  This is the 
                                                                     # url you use to access canvas.  
                                                                     # The %s part is used to plug in the endpoints 
                                                                     # later in the script.

define('ACCESS_TOKEN', "<your_access_token>"); # Replace this with your user's access token

$course_id = <course_id>;  # not the sis_id but the canvas internal id

// End of settings that would be changed

$request_headers = array('Authorization: Bearer ' . ACCESS_TOKEN);


# First, get the list of students in the course
#
# Build the API endpoint for fetching the list of students
$students_endpoint = sprintf(BASE_URL,sprintf('/courses/%d/students', $course_id));

# Initiate cURL, adding the REQUEST_HEADERS to it for authentication
$ch = curl_init();

// Set headers
curl_setopt($ch,CURLOPT_HTTPHEADER,$request_headers);

// Set url
curl_setopt($ch, CURLOPT_URL,$students_endpoint);
// Tell cURL to return the results of the request. 
curl_setopt($ch,CURLOPT_RETURNTRANSFER,True);
//
# Execute the request
# Fetch the response, 
$student_list = curl_exec($ch);

$grades = False;
if(!$student_list){
  print 'error fetching student list';
}else{
  # Strip extra whitespace from the beginning and end then parse response into JSON
  $student_list = json_decode(trim($student_list));

  # Build the query parameters, creating a student_ids[]=<id> for each student id
  $submission_params = array('include[]=total_scores','grouped=1');
  foreach($student_list as $key=>$value){
    $submission_params[] = 'student_ids[]='.$value->id;
  }
  # implode the string, joining it with '&'
  $submission_params_string = implode('&',$submission_params);

  # Build a request url 
  $submissions_endpoint = sprintf(BASE_URL,sprintf('/courses/%d/students/submissions', $course_id));
  # add the query parameters to the url, and set the curl url destination
  curl_setopt($ch, CURLOPT_URL, $submissions_endpoint . '?' . $submission_params_string);

  # Fetch the response
  # Get the resutls
  $grades = curl_exec($ch);

  if(!$grades){
    print "problem fetching submissions\n";
  }else{
    # strip extra whitespace from the beginning and end
    $grades = trim($grades);
    # Load the response as JSON
    $grades = json_decode($grades);
  }
}

# Please close the curl handler
curl_close($ch);

// Do something with the $grades array, which now contains the grades
var_dump($grades);
